### PR TITLE
Update datatypes_conf.xml.sample with docx type

### PR DESCRIPTION
**Pull request by yvanlebras**
(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by yvanlebras** [comment]
and in ue upstream galaxy code https://github.com/usegalaxy-eu/galaxy/pull/192

**Comment by yvanlebras** [comment]
New try

File: lib/galaxy/config/sample/datatypes_conf.xml.sample [hunk 1]
```diff
@@ -567,6 +567,8 @@
     <!-- Excel datatypes -->
     <datatype extension="excel.xls" type="galaxy.datatypes.binary:ExcelXls" display_in_upload="true"/>
     <datatype extension="xlsx" type="galaxy.datatypes.binary:Xlsx" display_in_upload="true"/>
+    <!-- Word datatypes -->
+    <datatype extension="docx" type="galaxy.datatypes.binary:Docx" display_in_upload="true"/>
```
Comment (bernt-matthias) [inline lib/galaxy/config/sample/datatypes_conf.xml.sample]: Why display in upload?

File: lib/galaxy/datatypes/binary.py [hunk 2]
```diff
@@ -4540,3 +4540,11 @@ def display_peek(self, dataset: DatasetProtocol) -> str:
             return dataset.peek
         except Exception:
             return f"Binary FITS file size ({nice_size(dataset.get_size())})"
+
+class Docx(Binary):
```
Comment (bernt-matthias) [inline lib/galaxy/datatypes/binary.py]: Not necessary. Just refer to Binary in the datatypes xml.

**Review by bernt-matthias — COMMENTED** [review]
Wondering where this is needed. IMHO docx is a format that should be avoided.

File: lib/galaxy/datatypes/binary.py (hunk 2 above)
Comment (yvanlebras) [reply lib/galaxy/datatypes/binary.py]: ok, thank you!

File: lib/galaxy/config/sample/datatypes_conf.xml.sample (hunk 1 above)
Comment (yvanlebras) [reply lib/galaxy/config/sample/datatypes_conf.xml.sample]: you're right! THANK YOU

**Comment by yvanlebras** [comment]
You right Matthias BUT users / reaserchers want to have a docx in our case (generation of a draft of data paper from a metadata file) so they can download it to edit it before submitting it....

**Review by bernt-matthias — APPROVED** [review]

File: lib/galaxy/config/sample/datatypes_conf.xml.sample [hunk 3]
```diff
@@ -567,6 +567,8 @@
     <!-- Excel datatypes -->
     <datatype extension="excel.xls" type="galaxy.datatypes.binary:ExcelXls" display_in_upload="true"/>
     <datatype extension="xlsx" type="galaxy.datatypes.binary:Xlsx" display_in_upload="true"/>
+    <!-- Word datatypes -->
+    <datatype extension="docx" type="galaxy.datatypes.binary:Binary"/>
```
Comment (bgruening) [inline lib/galaxy/config/sample/datatypes_conf.xml.sample]:
```suggestion
    <datatype extension="docx" type="galaxy.datatypes.binary:Binary" subclass="true"/>
```

**Comment by yvanlebras** [comment]
Hi Mattias, Björn, do you think this can be merged ?

**Review by bgruening — APPROVED** [review]

**Pull request merged**